File: 2021/01/24/minhyungPark/algorithm/LeetCode1725.kt
import kotlin.math.min

class LeetCode1725 {
    fun countGoodRectangles(rectangles: Array<IntArray>): Int {
        val squareLengths = rectangles.map { min(it[0], it[1]) }
        val maxLen = squareLengths.max()!!
        return squareLengths.filter { it == maxLen }.size
    }
}

File: 2021/01/31/1ilsang/algorithm/leetcode1725.js
/**
 * @param {number[][]} rectangles
 * @return {number}
 */
const countGoodRectangles = (rectangles) => {
  const squareLenArr = []
  rectangles.forEach( rect => {
      squareLenArr.push(Math.min(...rect))
  });
  const largest = Math.max(...squareLenArr)
  return squareLenArr.filter( l => l == largest ).length
};